### PR TITLE
feat(api): migration remove multiple spaces from filename (applics-1598)

### PR DESCRIPTION
**Pull request by DavidJWhiteDev**
## Describe your changes

When migrating documents, the document title is created by taking the doc title from HZN, and stripping off any know file suffix - to be compatible with existing CBOS functionality.  This PR also removes any multiple spaces from the title, and also trims.  This is consistent with what old NSIP website did when it received published docs.

## APPLICS-1598 Migration - Remove multiple spaces from filename
https://pins-ds.atlassian.net/browse/APPLICS-1598

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/api/src/server/utils/__tests__/file-fns.test.js [hunk 1]
```diff
@@ -30,4 +30,39 @@ describe('trimDocumentNameKnownSuffix', () => {
...
+		});
+
+		it('should replace all whitespace characters with single spaces', () => {
+			expect(removeMultipleSpacesAndTrim('a\tb\nc\rd')).toBe('a b c d');
+		});
+		it('should return empty string for empty input', () => {
+			// @ts-ignore
+			expect(removeMultipleSpacesAndTrim('')).toBe('');
+		});
+		// add test for null input
```
Comment (cprieststephens) [inline apps/api/src/server/utils/__tests__/file-fns.test.js:62]: Does this comment need to be removed?

**Review by cprieststephens — CHANGES_REQUESTED** [review]
